components/src/core/drawer/drawer-nav-item.tsx

```diff
@@ -85,7 +94,14 @@ export const DrawerNavItem: React.FC<DrawerNavItemProps> = (props) => {
                     <InfoListItem
                         dense
                         {...navItem}
-                        rightComponent={rightIcon}
+                        rightComponent={
+                            <View style={{ flexDirection: 'row', alignItems: 'center' }}>
+                                {navItem.rightComponent}
+                                {rightIcon && (
+                                    <View style={[defaultStyles.expandIcon, styles.expandIcon]}>{rightIcon}</View>
+                                )}
+                            </View>
+                        }
```

Comment:
Let's only render something if we have a right component or expand icon:
```suggestion
rightComponent={
                            (navItem.rightComponent || rightIcon) &&
                            <View style={{ flexDirection: 'row', alignItems: 'center' }}>
                                {navItem.rightComponent}
                                {rightIcon && (
                                    <View style={[defaultStyles.expandIcon, styles.expandIcon]}>{rightIcon}</View>
                                )}
                            </View>
                        }
```